xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/editBus.js

```diff
@@ -0,0 +1,232 @@
+     */
+    save(entryId) {
+      const values = this.editStates[entryId];
+      var canBeSaved = false;
+      var keyEntry = undefined;
+
+
+      // Look for the single cell to save.
+      for (keyEntry in values) {
+        const entryValue = values[keyEntry];
+
+        const editing = entryValue.editing;
+        const tosave = entryValue.tosave;
+        canBeSaved = !editing && tosave;
+
+        if (canBeSaved) break;
+      }
+
+      // If a cell to save is found, we aggegrate the content to save and save it. 
+      if (canBeSaved && keyEntry) {
+        // Aggregates the content of the form values of each property.
+
+        const vals = {};
+        values[keyEntry].content.forEach((content) => {
+          aggregate(content, vals);
+        })
+
+        this.logic.setValues({entryId, values: vals})
+          .then(() => {
+            this.editStates[entryId] = {};
+          })
+          .catch(() => {
+            new XWiki.widgets.Notification(`The row save action failed.`, 'error');
+          })
+      }
+    }
+
+    isEditable() {
+      for (const editStatesKey in this.editStates) {
+        const editStates = this.editStates[editStatesKey];
+        for (const editStateKey in editStates) {
+          const editState = editStates[editStateKey];
+          if (editState.editing) {
+            return false;
+          }
+        }
+      }
+      return true;
+    }
+  }
+
+  /**
+   * Initialized the edit bus view and services.
+   * @param logic the live data logic instance
+   */
+  function init(logic) {
+    _logic = logic
+    _editBusService = new EditBusService(_editBus, logic);
+    _editBusService.init()
+  }
+
+  /**
+   * Notifies the start of a cell modification. After this event, the cell is considered as edited unless it is canceled.
+   * @param entry the entry of the edited row
+   * @param propertyId the property id of the edited cell.
+   */
+  function start(entry, propertyId) {
+    _editBus.$emit('start-editing-entry', {
+      entryId: _logic.getEntryId(entry),
+      propertyId
+    });
+
+  }
+
+  /**
+   * Notifies the cancellation of the edition of a cell. The cell rollback to a non modified state.
+   * @param entry the entry of the edited row
+   * @param propertyId the property id of the edited cell
+   */
+  function cancel(entry, propertyId) {
+    _editBus.$emit('cancel-editing-entry', {
+      entryId: _logic.getEntryId(entry),
+      propertyId
+    })
+
+  }
+
+  /**
+   * Notifies the save of the a cell. With the current save strategy, the cell is directly save and the table is reload
+   * after this notification.
+   * @param entry the entry of the edit row
+   * @param propertyId the property id of the edited cell
+   * @param content the content of the edited cell, an array of objects with a single key and value pair,
+   *                for instance `[{"staticList1":"comedy"},{"staticList1":"romance"},{"staticList1":""}]
```

Comment:
The format of the ``content`` parameter is unexpected and unnatural. I don't see why it has to be so complicated.